numpy/core/src/multiarray/descriptor.c

```diff
@@ -3422,7 +3433,7 @@ NPY_NO_EXPORT PyTypeObject PyArrayDescr_Type = {
     arraydescr_methods,                         /* tp_methods */
     arraydescr_members,                         /* tp_members */
     arraydescr_getsets,                         /* tp_getset */
-    0,                                          /* tp_base */
+    BASETYPE,                                   /* tp_base */
```

Comment:
Based on discussion elsewhere, this _might_ need to be PyHeapType_Type. Will assume I'm wrong until something proves otherwise though.

Reply:
I will add a comment